docs/DeveloperGuide.md

```diff
@@ -151,12 +159,23 @@ The `Employee` component stores an employee's data which comprises:
 
 The `Project` component stores a project's data which comprises:
 
-* a `name` (stored as a String), `Deadline`, `UniqueEmployeeList`, `ProjectPriority` and `CompletionStatus`.
+* a `Name`, `Deadline`, `UniqueEmployeeList`, `TaskList`, `ProjectPriority` and `CompletionStatus`.
 * The `Deadline` is optional.
 * The `UniqueEmployeeList` contains only the `Employee`s under the said `Project`.
+* The `TaskList` contains the `Task`s which have been added to the `Project`.
 
 ***Note: For the Model, Employee and Project components, lower-level details (e.g. most class attributes and methods) have been omitted for visual clarity.
 
+### Task component
+**API** : [`Task.java`](https://github.com/AY2324S1-CS2103T-T08-3/tp/blob/master/src/main/java/seedu/address/model/task/Task.java)
+
+<img src="images/TaskClassDiagram.png" width="450" />
+
+The `Task` component stores a task's data which comprises:
+
+* a `name`, `deadline`, `employee`, and `isDone` field.
```

Comment:
For consistency with the rest of the DG, I think it would be good if all class names were in UpperCamelCase (Employee in this case). Regardless, I will fix anything related to the Model section in my PR since I was working on it earlier.